### PR TITLE
feat(rest.network.configuration.provider): new rest api for network configuration management

### PR DESCRIPTION
**Pull request by sfiorani**
This PR add a new REST api for managing network configuraion.

**Related Issue:**

**Description of the solution adopted:**

**Screenshots:**

**Manual Tests:**

**Any side note on the changes made:**

### CODE REVIEW

File: kura/org.eclipse.kura.rest.network.configuration.provider/src/main/java/org/eclipse/kura/internal/rest/network/configuration/NetworkConfigurationRestService.java [hunk 1]
```diff
@@ -0,0 +1,278 @@
...
+import org.osgi.service.useradmin.UserAdmin;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Path("networkConfiguration/v1")
+public class NetworkConfigurationRestService {
+
+    private static final Logger logger = LoggerFactory.getLogger(NetworkConfigurationRestService.class);
+
+    private static final String APP_ID = "NETCONF-V1";
```
Comment (nicolatimeus) [inline kura/org.eclipse.kura.rest.network.configuration.provider/src/main/java/org/eclipse/kura/internal/rest/network/configuration/NetworkConfigurationRestService.java]: No need to expose this as a request handler, please remove this.

File: kura/org.eclipse.kura.rest.network.configuration.provider/src/main/java/org/eclipse/kura/internal/rest/network/configuration/NetworkConfigurationRestService.java [hunk 2]
```diff
@@ -0,0 +1,278 @@
...
+    private static final String NETWORK_CONF_SERVICE_PID = "org.eclipse.kura.net.admin.NetworkConfigurationService";
+    private static final String IP4_FIREWALL_CONF_SERVICE_PID = "org.eclipse.kura.net.admin.FirewallConfigurationService";
+    private static final String IP6_FIREWALL_CONF_SERVICE_PID = "org.eclipse.kura.net.admin.ipv6.FirewallConfigurationServiceIPv6";
+
+    private static final List<String> NETWORK_CONFIGURATION_PIDS = Arrays.asList(NETWORK_CONF_SERVICE_PID,
+            IP4_FIREWALL_CONF_SERVICE_PID, IP6_FIREWALL_CONF_SERVICE_PID);
+
+    private static final String SNAPSHOT_SUBTASK_ID = "snapshot";
+
+    private final RequestHandler requestHandler = new JaxRsRequestHandlerProxy(this);
```
Comment (nicolatimeus) [inline kura/org.eclipse.kura.rest.network.configuration.provider/src/main/java/org/eclipse/kura/internal/rest/network/configuration/NetworkConfigurationRestService.java]: No need to expose this as a request handler, please remove this.

File: kura/org.eclipse.kura.rest.network.configuration.provider/src/main/java/org/eclipse/kura/internal/rest/network/configuration/NetworkConfigurationRestService.java [hunk 3]
```diff
@@ -0,0 +1,278 @@
...
+
+    public void setConfigurationService(ConfigurationService configurationService) {
+        this.configurationService = configurationService;
+    }
+
+    public void setCryptoService(CryptoService cryptoService) {
+        this.cryptoService = cryptoService;
+    }
+
+    public void setRequestHandlerRegistry(final RequestHandlerRegistry registry) {
```
Comment (nicolatimeus) [inline kura/org.eclipse.kura.rest.network.configuration.provider/src/main/java/org/eclipse/kura/internal/rest/network/configuration/NetworkConfigurationRestService.java]: No need to expose this as a request handler, please remove this.

File: kura/org.eclipse.kura.rest.network.configuration.provider/src/main/java/org/eclipse/kura/internal/rest/network/configuration/NetworkConfigurationRestService.java [hunk 4]
```diff
@@ -0,0 +1,278 @@
...
+        try {
+            registry.registerRequestHandler(APP_ID, this.requestHandler);
+        } catch (final Exception e) {
+            logger.warn("failed to register request handler", e);
+        }
+    }
+
+    public void unsetRequestHandlerRegistry(final RequestHandlerRegistry registry) {
```
Comment (nicolatimeus) [inline kura/org.eclipse.kura.rest.network.configuration.provider/src/main/java/org/eclipse/kura/internal/rest/network/configuration/NetworkConfigurationRestService.java]: No need to expose this as a request handler, please remove this.

File: kura/org.eclipse.kura.rest.network.configuration.provider/OSGI-INF/network_configuration_rest_service.xml [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+	<property name="kura.service.pid" type="String"
+		value="org.eclipse.kura.internal.rest.network.configuration.NetworkConfigurationRestService" />
+
+	<reference bind="setConfigurationService" cardinality="1..1"
+		interface="org.eclipse.kura.configuration.ConfigurationService"
+		name="ConfigurationService" policy="static" />
+	<reference bind="setUserAdmin" cardinality="0..1"
+		interface="org.osgi.service.useradmin.UserAdmin" name="UserAdmin"
+		policy="static" />
+	<reference bind="setRequestHandlerRegistry" cardinality="0..n"
```
Comment (nicolatimeus) [inline kura/org.eclipse.kura.rest.network.configuration.provider/OSGI-INF/network_configuration_rest_service.xml]: No need to expose this as a request handler, please remove this.

File: kura/org.eclipse.kura.rest.network.configuration.provider/src/main/java/org/eclipse/kura/internal/rest/network/configuration/NetworkConfigurationRestService.java [hunk 6]
```diff
@@ -0,0 +1,278 @@
...
+     *
+     * Lists the tracked network configurable component Pids
+     *
+     * @return a List of String objects representing the Pids of network factory components
+     *         tracked by the
+     *         {@link ConfigurationService}
+     */
+    @GET
+    @RolesAllowed("network.configuration")
+    @Path("/networkConfigurableComponents")
```
Comment (nicolatimeus) [inline kura/org.eclipse.kura.rest.network.configuration.provider/src/main/java/org/eclipse/kura/internal/rest/network/configuration/NetworkConfigurationRestService.java]: Maybe you can replace `networkConfigurableComponents` with `configurableComponents` to be consistent with CONF-V2, we already have `networkConfiguration` in the prefix

**Comment by nicolatimeus** [comment]
https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/11344

**Comment by sfiorani** [comment]
Tested on Raspberry Pi with generic-aarch64 kura profile

**Comment by nicolatimeus** [comment]
There are conflicts that need to be solved

File: kura/org.eclipse.kura.rest.network.configuration.provider/build.properties [hunk 7]
```diff
@@ -0,0 +1,19 @@
+#
+#  Copyright (c) 2021 Eurotech and/or its affiliates and others
```
Comment (nicolatimeus) [inline kura/org.eclipse.kura.rest.network.configuration.provider/build.properties]: Set to 2023

File: kura/test/org.eclipse.kura.rest.network.configuration.provider.test/build.properties [hunk 8]
```diff
@@ -0,0 +1,15 @@
+#
+#  Copyright (c) 2021 Eurotech and/or its affiliates and others
```
Comment (nicolatimeus) [inline kura/test/org.eclipse.kura.rest.network.configuration.provider.test/build.properties]: 2023

File: kura/test/org.eclipse.kura.rest.network.configuration.provider.test/src/main/java/org/eclipse/kura/rest/network/configuration/provider/test/responses/MockComponentConfiguration.java [hunk 9]
```diff
@@ -0,0 +1,74 @@
+package org.eclipse.kura.rest.network.configuration.provider.test.responses;
```
Comment (nicolatimeus) [inline kura/test/org.eclipse.kura.rest.network.configuration.provider.test/src/main/java/org/eclipse/kura/rest/network/configuration/provider/test/responses/MockComponentConfiguration.java:13]: Copyright header is missing

File: kura/test/org.eclipse.kura.rest.network.configuration.provider.test/src/main/java/org/eclipse/kura/rest/network/configuration/provider/test/responses/RestNetworkConfigurationJson.java [hunk 10]
```diff
@@ -0,0 +1,19 @@
+package org.eclipse.kura.rest.network.configuration.provider.test.responses;
```
Comment (nicolatimeus) [inline kura/test/org.eclipse.kura.rest.network.configuration.provider.test/src/main/java/org/eclipse/kura/rest/network/configuration/provider/test/responses/RestNetworkConfigurationJson.java:13]: Copyright header is missing

**Comment by nicolatimeus** [comment]
This bundle should only be added to the distribution profiles with network management support, it is not needed on the `-nn` profiles. Please remove the new permission from the `snapshot_0.xml` of nn profiles and modify the ant script to add the new bundle only if we are not building an `-nn` profile.

**Comment by sfiorani** [comment]
Eclipse team approved changes

**Review by nicolatimeus — APPROVED** [review]

**Pull request merged**